homeassistant/components/bsblan/coordinator.py

```diff
@@ -0,0 +1,56 @@
+"""DataUpdateCoordinator for the BSB-Lan integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+from random import randint
+
+from bsblan import BSBLAN, BSBLANConnectionError
+from bsblan.models import State
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, LOGGER, SCAN_INTERVAL
+
+
+class BSBLanUpdateCoordinator(DataUpdateCoordinator[State]):
+    """The BSB-Lan update coordinator."""
+
+    config_entry: ConfigEntry
+    client: BSBLAN
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: ConfigEntry,
```

Comment:
```suggestion
```

Reply:
Yes, but this is needed for L37 otherwise is `config_entry` only available after the `super().__init__()`